packages/affiliates/libs/uniswap/processor.ts

```diff
@@ -0,0 +1,249 @@
+// this file is meant to handle events from various contracts.
+// its mainly documenting all events based on uniswap typechain build output.
+import { Positions, NftPositions, Pools, Position } from "./models";
+import { exists, convertValuesToString } from "./utils";
+import assert from "assert";
+import { BigNumberish } from "ethers";
+
+type Event = {
+  event: string;
+  args: any[];
+};
+
+type PoolsType = ReturnType<typeof Pools>;
+
+export function NftEvents({ positions }: { positions: ReturnType<typeof NftPositions> }) {
+  const handlers: { [key: string]: (...args: any[]) => Promise<void> } = {
+    async Approval(owner: string | null, approved: string | null, tokenId: BigNumberish | null) {
+      // nothing
+    },
+    async ApprovalForAll(owner: string | null, operator: string | null, approved: null) {
+      // nothing
+    },
+    async Collect(
+      tokenId: BigNumberish | null,
+      recipient: string | null,
+      amount0: BigNumberish | null,
+      amount1: BigNumberish | null
+    ) {
+      // nothing
+    },
+    async DecreaseLiquidity(
+      tokenId: BigNumberish | null,
+      liquidity: BigNumberish | null,
+      amount0: BigNumberish | null,
+      amount1: BigNumberish | null
+    ) {
+      assert(exists(tokenId), "requires token id");
```

Comment:
Instead of `asserting`  that these params are not `null`, why can't you just remove the `| null` from the type?

Reply:
This was actually generated by typechain based on the solidity code. I found it weird that these values could be null, and maybe in practice they would never be? There were instances where the generated typescript was just null for some reason, so i think maybe its not bullet proof and could just remove the nulls